CMakeLists.txt

```diff
@@ -13,7 +41,11 @@ if(USE_OPENMP)
 endif()
 
 # System specific definitions
-include(CMake.system)
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/CMake.system")
+    message("Including user-provided CMake.system")
+    include("${CMAKE_CURRENT_SOURCE_DIR}/CMake.system")
+endif()
```

Comment:
This check should be done above. If CMake.system exists, then use the definitions therein; only try searching for Armadillo and GSL if CMake.system does not exist.